waspc/src/Wasp/Generator/Valid/Validator.hs

```diff
@@ -0,0 +1,73 @@
+module Wasp.Generator.Valid.Validator where
+
+import Data.Bifunctor (first)
+import Data.Functor (void)
+import Data.List (intercalate)
+import Data.List.NonEmpty (NonEmpty)
+import qualified Data.List.NonEmpty as NE
+import Data.Maybe (catMaybes)
+import qualified Validation as V
+import Wasp.Util (indent)
+
```

Comment:
So here I would be tempted for us to do `type Validator input result = input -> Validation result`. Maaaaaybe even go as for as make it `data`, but likely not as we don't need branding and it would just complciate things.

Reply:
I already had this and I undid it at some point as I thought it would be introducing too many concepts 😅
I'll put it back in